### PR TITLE
H100 nodes: specs and slurm config

### PR DESCRIPTION
**Pull request by willfurnass**
- [x] Hardware specs
- [x] Slurm config

### CODE REVIEW

**Comment by willfurnass** [comment]
Not to be merged until just before H100 nodes made available to end users.

**Review by Carldkennedy — APPROVED** [review]
LGTM

**Review by nikishe — APPROVED** [review]
LGTM

File: stanage/cluster_specs.rst [hunk 1]
```diff
@@ -129,14 +156,14 @@ Total capacity
...
 
   * Intel Cores: 11264.
-  * AMD Cores: 768.
+  * AMD Cores: 1344.
   
-* Total memory:  83968 GiB.
-* GPUs: 64.
+* Total memory:  90112 GiB.
+* GPUs: 76.
 * Fast network filesystem (`Lustre`_):  2 PiB.
```
Comment (jkwmoore) [inline stanage/cluster_specs.rst:167]: Can we add an invisible RST comment here on the calculation method we're using for nodes, cores, memory etc... e.g. with / without GPU nodes, with/without "private nodes" etc...

Comment (willfurnass) [reply stanage/cluster_specs.rst:167]: Good idea; can handle via separate Issue/PR.

Comment (willfurnass) [reply stanage/cluster_specs.rst:167]: https://github.com/rcgsheffield/sheffield_hpc/issues/1795

**Review by jkwmoore — APPROVED** [review]

**Comment by willfurnass** [comment]
- [ ] Add info on NVlink vs PCIe: H100s might not always be faster
       - NVLink + 2x A100 gives 3-4x inter-GPU bandwidth vs PCIe + 2x H100
       - but inter-GPU latencies actually lower for 2x H100

**Comment by willfurnass** [comment]
Rebased on `master`

**Comment by willfurnass** [comment]
@Carldkennedy and @nikishe: Please can you review the new content on limitations of H100 support in latest PyTorch and Tensorflow releases: 

```
referenceinfo/imports/software/tensorflow/compat-table-import.rst
stanage/software/apps/pytorch.rst
stanage/software/apps/tensorflow.rst
```

**Comment by willfurnass** [comment]
Also, where should we stick comments about perf differences between H100s and A100s being greatest when probs are compute-bound (rather than mem-bound) and with larger prob sizes?

File: stanage/software/apps/pytorch.rst [hunk 2]
```diff
@@ -48,17 +53,19 @@ Then PyTorch can be installed by the following ::
...
 
    # Activate the 'pytorch' environment
    source activate pytorch
 
-   # Install PyTorch
-   pip install torch torchvision
+   # Install the latest stable PyTorch release if you only want to run PyTorch using A100 GPUs
+   python -m pip install torch torchvision
+   # Or use a nightly PyTorch build instead if you want to be able to use H100 GPUs also / as well.
+   # python -m pip install --pre torch torchvision --index-url https://download.pytorch.org/whl/nightly/cu118
```
Comment (Carldkennedy) [inline stanage/software/apps/pytorch.rst:68]: `module load cuDNN/8.7.0.84-CUDA-11.8.0` for pytorch-nightly build?

**Review by Carldkennedy — COMMENTED** [review]
LGTM other than one comment

**Comment by Carldkennedy** [comment]
> Also, where should we stick comments about perf differences between H100s and A100s being greatest when probs are compute-bound (rather than mem-bound) and with larger prob sizes?

Maybe a note imported in GPU computing and below example batch scripts in  R, lammps, gromacs, pytorch and tensorflow?

**Comment by jkwmoore** [comment]
I've amended slightly to fully specify the GPU GRES with a mind to us 

A: sticking to this approach in future to ensure user's decouple/select GPUs correctly.
B: can add a "Any GPU node(s)" option, when the H100s are moved into the main GPU partition.

With respect to Bessemer and the temporarily available  V100 16GB models, we can then be opinionated via `--gres=gpu:v100-16GB:1` or similar.

We may wish to align the release of the H100 with any amendment to the Bessemer setup in order to use the same notification email (but since any changes to Bessemer are a bit of a breaking change we need to give a week's notice!)

**Comment by jkwmoore** [comment]
I have amended this to add explicit warnings about our "opt-in" plan of action.

**Review by Carldkennedy — APPROVED** [review]
LGTM

**Pull request merged**